pkg/cmd/historical_logs.go

```diff
@@ -69,78 +86,166 @@ func NewCmdLogFilter(streams genericclioptions.IOStreams) *cobra.Command {
+		if err != nil {
+			return err
+		}
+
+		err = printLogs(logList, streams, o.Limit)
+
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func fetchLogs(logList *[]string, baseUrl string, logParameters *LogParameters) error {
+
+	req, err := http.NewRequest("GET", baseUrl, nil)
+	if err != nil {
+		return fmt.Errorf("http request failed: %v", err)
+	}
+
+	query := req.URL.Query()
+	query.Add("podname", logParameters.Podname)
+	query.Add("namespace", logParameters.Namespace)
+	query.Add("starttime", logParameters.StartTime)
+	query.Add("finishtime", logParameters.EndTime)
+	query.Add("maxlogs", logParameters.Limit)
+	query.Add("level", logParameters.Level)
+	req.URL.RawQuery = query.Encode()
 
-	req, err := http.NewRequest("GET", ApiUrl, payload)
+	response, err := http.DefaultClient.Do(req)
 
 	if err != nil {
-		return fmt.Errorf("Request Failed: %v", err)
+		return fmt.Errorf("failed to get http response %v", err)
 	}
 
-	client := &http.Client{}
-	res, err := client.Do(req)
+	responseBody, err := ioutil.ReadAll(response.Body)
 	if err != nil {
-		return fmt.Errorf("Response Error: %v", err)
+		return fmt.Errorf("failed to read response: %v", err)
 	}
 
-	responseBody, err := ioutil.ReadAll(res.Body)
+	err = response.Body.Close()
+
 	if err != nil {
-		return fmt.Errorf("Failed to read Response: %v", err)
+		return fmt.Errorf("an error occurred while attempting to close response body %v", err)
 	}
 
 	jsonResponse := &ResponseLogs{}
-	json.Unmarshal(responseBody, &jsonResponse)
+	err = json.Unmarshal(responseBody, &jsonResponse)
+	if err != nil {
+		return fmt.Errorf("an error occurred while unmarshalling JSON response: %v", err)
+	}
 
-	for i := 0; i < len(jsonResponse.Logs); i++ {
+	for _, log := range jsonResponse.Logs {
 
-		log := LogOptions{}
-		err = json.Unmarshal([]byte(jsonResponse.Logs[i]), &log)
+		logOption := logs.LogOptions{}
+		err := json.Unmarshal([]byte(log), &logOption)
 
 		if err != nil {
-			return fmt.Errorf("An Error Occurred while decoding response: %v", err)
+			return fmt.Errorf("no logs present, or input parameters were invalid")
 		}
 
-		if len(log.Source.Message) > 0 {
-			fmt.Fprintf(streams.Out, log.Source.Message+"\n")
+		if len(logOption.Source.Message) > 0 {
+			if logParameters.Prefix && len(logOption.Source.Kubernetes.PodName) > 0 && len(logOption.Source.Kubernetes.ContainerName) > 0 {
+				*logList = append(*logList, "pod/"+logOption.Source.Kubernetes.PodName+"/"+logOption.Source.Kubernetes.ContainerName+"   "+logOption.Source.Message)
+			} else {
+				*logList = append(*logList, logOption.Source.Message)
+			}
 		}
+	}
 
+	return nil
+}
+
+func printLogs(logList []string, streams genericclioptions.IOStreams, limit string) error {
+
+	if len(logList) == 0 {
+		return fmt.Errorf("no logs present, or input parameters were invalid")
+	}
+
+	var maxLogs int
+	var err error
+
+	if len(limit) > 0 {
+		maxLogs, err = strconv.Atoi(limit)
+		if err != nil || maxLogs < 0 || maxLogs > 1000 {
```

Comment:
Also, it would be nice to move this 1000 to a constant and then use it here and in the error message below. It would be easier to change it in one place in the future.